ui/src/components/TrafficPage.tsx

```diff
@@ -272,19 +276,51 @@ export const TrafficPage: React.FC<TrafficPageProps> = ({onTLSDetected, setAnaly
         }
     }
 
+    useEffect(() => {
+        (async () => {
+            try {
+                const serviceMapStatusResponse = await api.serviceMapStatus();
+                if (serviceMapStatusResponse["status"] === "enabled") {
+                    setServiceMapStatus(true);
+                } 
+            } catch(error) {
+                console.error(error);
+            }
+        })()
+    }, []);
+
+    const openServiceMapModal = debounce(() => {
+        setServiceMapModalOpen(true)
+    }, 500);
+
     return (
         <div className="TrafficPage">
             <div className="TrafficPageHeader">
-                <img className="playPauseIcon" style={{visibility: connection === ConnectionStatus.Connected ? "visible" : "hidden"}} alt="pause"
-                    src={pauseIcon} onClick={toggleConnection}/>
-                <img className="playPauseIcon" style={{position: "absolute", visibility: connection === ConnectionStatus.Connected ? "hidden" : "visible"}} alt="play"
-                    src={playIcon} onClick={toggleConnection}/>
+                <img className="playPauseIcon" style={{ visibility: connection === ConnectionStatus.Connected ? "visible" : "hidden" }} alt="pause"
+                    src={pauseIcon} onClick={toggleConnection} />
+                <img className="playPauseIcon" style={{ position: "absolute", visibility: connection === ConnectionStatus.Connected ? "hidden" : "visible" }} alt="play"
+                    src={playIcon} onClick={toggleConnection} />
                 <div className="connectionText">
                     {getConnectionTitle()}
                     <div className={"indicatorContainer " + getConnectionStatusClass(true)}>
-                        <div className={"indicator " + getConnectionStatusClass(false)}/>
+                        <div className={"indicator " + getConnectionStatusClass(false)} />
                     </div>
                 </div>
+                <Button
+                    variant="contained"
+                    style={{
+                        visibility: serviceMapStatus === true ? "visible" : "hidden",
```

Comment:
you can do: serviceMapStatus ? "visible" : "hidden"

Reply:
and anyway I think it is better to do something like {serviceMapStatus && <Button...}